connection.go

```diff
@@ -106,10 +98,7 @@ func (c *Connection) Close() error {
 // This function is used internally by Run which should be used for most queries.
 func (c *Connection) Query(q Query) (*Response, *Cursor, error) {
 	c.mu.Lock()
-	if c == nil {
```

Comment:
That sounds good.  I'll also remove the `c.mu.Unlock()` inside of this condition since that will panic.